fsspec/implementations/local.py

```diff
@@ -206,24 +211,32 @@ def modified(self, path):
 
     @classmethod
     def _parent(cls, path):
-        path = cls._strip_protocol(path).rstrip("/")
-        if "/" in path:
-            return path.rsplit("/", 1)[0]
+        path = cls._strip_protocol(path, remove_trailing_slash=True)
+        if os.sep == "/":
+            # posix native
+            return path.rsplit("/", 1)[0] or "/"
         else:
-            return cls.root_marker
+            # NT
+            path_ = path.rsplit("/", 1)[0]
+            if len(path_) <= 3:
+                if path_[1:2] == ":":
+                    # nt root (something like c:/)
+                    return path_[0] + ":/"
+            # More cases may be required here
+            return path_
 
     @classmethod
-    def _strip_protocol(cls, path):
+    def _strip_protocol(cls, path, remove_trailing_slash=False):
         path = stringify_path(path)
-        if path.startswith("file://"):
-            path = path[7:]
-        elif path.startswith("file:"):
-            path = path[5:]
-        elif path.startswith("local://"):
-            path = path[8:]
+        if path.startswith("file:"):
+            path = _remove_prefix(_remove_prefix(path, "file://"), "file:")
```

Comment:
It is faster to use regex
```
pattern = re.compile("file:[//]?")
%timeit pattern.sub("", "file://my/path/file")
180ns
%timeit _remove_prefix(_remove_prefix("file://my/path/file", "file://"), "file:")
260ns
```

and you _could_ do all possible patterns in one
```
pattern = re.compile(f"({'|'.join(protocol)}):[//]?")
pattern.sub("", path)
```
(you want to compile just once, and then the sub part is faster)

Reply:
The first `timeit` isn't taking into account compiling the pattern. Would the pattern be compiled during loading as a global?

Reply:
I have no objections if you can re-write it as a regex.

Reply:
> Would the pattern be compiled during loading as a global?

Exactly what I was thinking. 

However, I am finding that for the case where the path passed is already stripped, your code is actually (just) faster, so no change needed.